pallets/fusion/src/lib.rs

```diff
@@ -0,0 +1,1989 @@
+
+			let account_id = format!("{}{}", POOL_ACCOUNT_PREFIX, pool_id);
+			let pool_account =
+				T::PalletId::get().into_sub_account_truncating(account_id.as_bytes());
+
+			let mut reward_account: Option<T::AccountId> = None;
+			if with_reward_account {
+				let reward_acc_id = format!("{}{}", POOL_REWARD_ACCOUNT_PREFIX, pool_id);
+				reward_account =
+					Some(T::PalletId::get().into_sub_account_truncating(reward_acc_id.as_bytes()));
+			}
+
+			let new_pool = FusionPool::<T> {
+				pool_id,
+				currency_id,
+				apy,
+				pool_account,
+				state: FusionPoolState::Paused,
+				nominator: nominator.clone(),
+				reward_account: reward_account.clone(),
+				members: BoundedVec::default(),
+				targets: BoundedVec::default(),
+				total_staked_native: 0,
+				total_staked_points: 0,
+				total_slashed_native: 0,
+				total_unbonding_native: 0,
+			};
+
+			FusionPools::<T>::insert(pool_id, new_pool);
+			FusionPoolCount::<T>::mutate(|count| *count += 1);
+
+			Self::deposit_event(Event::PoolCreated {
+				pool_id,
+				currency_id,
+				apy,
+				state: FusionPoolState::Paused,
+				nominator,
+				reward_account,
+			});
+
+			Ok(())
+		}
+
+		/// Updates an existing fusion pool
+		#[pallet::call_index(7)]
+		#[pallet::weight(T::WeightInfo::create_fusion_currency())]
+		pub fn set_fusion_pool(
+			origin: OriginFor<T>,
+			pool_id: PoolId,
+			apy: Option<Perbill>,
+			state: Option<FusionPoolState>,
+			nominator: Option<Option<T::AccountId>>,
+		) -> DispatchResult {
+			ensure_root(origin)?;
+
+			FusionPools::<T>::try_mutate(pool_id, |maybe_pool| {
+				let pool = maybe_pool.as_mut().ok_or(Error::<T>::PoolNotFound)?;
+
+				ensure!(
+					pool.state != FusionPoolState::Destroying,
+					Error::<T>::PoolIsDestroying
+				);
+
+				if let Some(apy_value) = apy {
+					pool.apy = apy_value;
+				}
+
+				if let Some(state) = state {
+					ensure!(
+						state != FusionPoolState::Destroying,
+						Error::<T>::CannotSetPoolToDestroying
+					);
+					if state == FusionPoolState::Open || state == FusionPoolState::Blocked {
+						ensure!(pool.targets.len() > 0, Error::<T>::PoolIsNotNominating);
+						let currency = FusionCurrencies::<T>::get(pool.currency_id)
+							.ok_or(Error::<T>::CurrencyNotFound)?;
+						ensure!(!currency.is_destroyed, Error::<T>::CurrencyDestroyed);
+					}
+
+					pool.state = state;
+				}
+
+				if let Some(nominator) = nominator.clone() {
+					pool.nominator = nominator;
+				}
+
+				// Emit an event for pool update
+				Self::deposit_event(Event::PoolSet {
+					pool_id,
+					apy,
+					state,
+					nominator,
+				});
+
+				Ok(())
+			})
+		}
+
+		/// Deletes a pool
+		/// Called once to set the pool to destroying
+		/// Called a second time when everything is cleaned to actually destroy it
+		#[pallet::call_index(8)]
+		#[pallet::weight(T::WeightInfo::create_fusion_currency())]
+		pub fn delete_fusion_pool(origin: OriginFor<T>, pool_id: PoolId) -> DispatchResult {
```

Comment:
TODO

Reply:
What do you mean ?

Reply:
It's a message for me to think about the naming scheme. `Remove` instead of `delete` might be a better name but I will need to do some research and see what nomenclature is used in polkadot sdk.

Reply:
Actually should be destroy i guess